tests/transformer/remove_comments_test.cpp

```diff
@@ -0,0 +1,86 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "../test.h"
+#include "transformer/remove_comments.hpp"
+#include "transformer_utils.hpp"
+
+TEST(TestRemoveComments, NameAndID)
+{
+    EXPECT_STREQ(transformer::remove_comments::name().data(), "remove_comments");
+    EXPECT_EQ(transformer::remove_comments::id(), transformer_id::remove_comments);
+}
+
+TEST(TestRemoveComments, EmptyString) { EXPECT_NO_TRANSFORM(remove_comments, ""); }
+
+TEST(TestRemoveComments, ValidTransformShellComment)
+{
+    EXPECT_TRANSFORM(remove_comments, "#", "");
+    EXPECT_TRANSFORM(remove_comments, "#foo", "");
+    EXPECT_TRANSFORM(remove_comments, "bar#foo", "bar");
+    EXPECT_TRANSFORM(remove_comments, "bar#", "bar");
+}
+
+TEST(TestRemoveComments, ValidTransformSQLComment)
+{
+    EXPECT_TRANSFORM(remove_comments, "--", "");
+    EXPECT_TRANSFORM(remove_comments, "--foo", "");
+    EXPECT_TRANSFORM(remove_comments, "bar--foo", "bar");
+    EXPECT_TRANSFORM(remove_comments, "bar--", "bar");
+}
+
+TEST(TestRemoveComments, ValidTransformCPPComment)
+{
+    EXPECT_TRANSFORM(remove_comments, "//", "");
```

Comment:
I don't have the whole background of transformers but would it be possible to get something like `//blaaaa\nNoCommentOnThisLine` ?

Reply:
In theory yes, but the remove comments transformer doesn't cover that case and I'm not sure if we want to cc: @Taiki-San